### PR TITLE
Add metrics for the bird info column

### LINKED ISSUE #1: Feature request: Parse Info column into prometheus metrics
Consider birdcl output like this :

```
name    proto    table    state  since      info
 Mesh_10_101_4_182 BGP      master   up     2016-10-23  Established
   Description:    Connection to BGP peer
   Preference:     100
   Input filter:   ACCEPT
   Output filter:  calico_pools
   Routes:         1 imported, 1 exported, 1 preferred
   Route change stats:     received   rejected   filtered    ignored   accepted
     Import updates:              1          0          0          0          1
     Import withdraws:            0          0        ---          0          0
     Export updates:             44         25         18        ---          1
     Export withdraws:            0        ---        ---        ---          0
   BGP state:          Established

```

Would you consider adding a prometheus line to the output that represents the info. 

For example 
bird_info{bird_protocol_instance="Mesh_10_101_4_182",value=unknown} 0 
bird_info{bird_protocol_instance="Mesh_10_101_4_182",value=Active} 0
bird_info{bird_protocol_instance="Mesh_10_101_4_182",value=Established} 1

I am willing to write the implementation and add the tests. 
I wanted to reach out to you first whether you would welcome a PR ?

### PR DESCRIPTION
**Pull request by ahakanbaba**
Also add tests.

Fixes #1

### CODE REVIEW

**Comment by ahakanbaba** [comment]
@EdSchouten Please take a look. Thanks.

File: birdwatcher_test.py [hunk 1]
```diff
@@ -12,11 +12,13 @@
 
 class TestParseShowProtocols(unittest.TestCase):
 
+    maxDiff = 4000
```
Comment (ahakanbaba) [inline birdwatcher_test.py:15]: WIth this change the unit test module shows the long diffs.

File: birdwatcher.py [hunk 2]
```diff
@@ -47,6 +52,11 @@ def parse_show_protocols(f):
         yield ('bird_up{bird_protocol_instance="%s"}' % protocol_instance,
                int(line[3] == 'up'))
 
+        info = line[5] if len(line)>5 else "unknown"
+        yield bird_info_line(protocol_instance, info, "unknown")
```
Comment (EdSchouten) [inline birdwatcher.py:56]: Looking at the Bird source code, there are more states than just Active and Established:

```
static char *bgp_state_names[] = { "Idle", "Connect", "Active", "OpenSent", "OpenConfirm", "Established", "Close" };
```

Maybe it makes more sense not to add any state names to the source code of this exporter, but simply to export just a single metric containing the literal value of the state?

Comment (ahakanbaba) [reply birdwatcher.py:56]: Thanks for the comment. 
How do you recommend doing that ? 

bird_info{{bird_protocol_instance="Mesh_10_101_4_114"} "Established" 
Is not a valid prometheus metric line. According to the documentation the value needs to be a float. 

[Here](https://prometheus.io/docs/instrumenting/exposition_formats/) 

> metric_name [
>   "{" label_name "=" `"` label_value `"` { "," label_name "=" `"` label_value `"` } [ "," ] "}"
> ] value [ timestamp ]
> 
"value is a float"

In terms of the generated metrics I see two options:

1) 
This is what this PR has. I can extend it with the other parameters in bird_state_names
bird_info{bird_protocol_instance="Mesh_10_101_4_114",value="unknown"}  0
bird_info{bird_protocol_instance="Mesh_10_101_4_114",value="Active"} 1
bird_info{bird_protocol_instance="Mesh_10_101_4_114",value="Established"} 0
2) I got the inspiration from [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics/blob/master/collectors/node.go#L256)  
bird_established{bird_protocol_instance="Mesh_10_101_4_114",condition="true"} 0
bird_established{bird_protocol_instance="Mesh_10_101_4_114",condition="false"} 1
bird_established{bird_protocol_instance="Mesh_10_101_4_114",condition="unknown"} 0
bird_active{bird_protocol_instance="Mesh_10_101_4_114",condition="true"} 1
bird_active{bird_protocol_instance="Mesh_10_101_4_114",condition="false"} 0
bird_active{bird_protocol_instance="Mesh_10_101_4_114",condition="unknown"} 0

In either case, I think the values in the bgp_state_names should be known in the birdwatcher.py implementation.

The birdwatcher script already knows about other parameters from bird. Is adding this a big problem ? 

Do you have any other option ? 

PS: I do not think emitting only the current value of the bird info ( the true value) is a viable option. 
For example if the only emitted metric is 
bird_info{bird_protocol_instance="Mesh_10_101_4_114",value="Active"} 1
How would the consumer of these metrics can trigger on a protocol not being Established? I am not sure whether there is a way to distinguish between a false value and a non-existing value.

Comment (ahakanbaba) [reply birdwatcher.py:56]: @EdSchouten any comments ?

Comment (EdSchouten) [reply birdwatcher.py:56]: Regarding your post scriptum: you can just use an alert like this, right?

```
ALERT Whatever
  IF bird_info{value!="Established"}
  ....
```

What are your thoughts on this change?

https://github.com/kumina/birdwatcher/compare/bgp_state

**Comment by ahakanbaba** [comment]
Thanks @EdSchouten the commit looks good. 
I think I may make our consumption side work. 

Please merge the bgp_state branch in favor of this PR. I am closing this. 

If for some reason that metric does not work for us I will let you know.

**Pull request closed without merging**